src/coffee/directives/api/utils/model-key.coffee

```diff
@@ -4,22 +4,36 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
     class ModelKey extends BaseObject
       constructor: (@scope) ->
         super()
+        ##common scope keys interface for iterating comparators
+        @interface = {}
+        @interface.scopeKeys = []
+
         @defaultIdKey = 'id'
         @idKey = undefined
 
       evalModelHandle: (model, modelKey) ->
-        if model == undefined or modelKey == undefined
-          return undefined
+        return if not model? or not modelKey?
+
         if modelKey == 'self'
           model
         else #modelKey may use dot-notation
+          modelKey = modelKey() if _.isFunction(modelKey)
           GmapUtil.getPath(model, modelKey)
 
       modelKeyComparison: (model1, model2) =>
-        scope = if @scope.coords? then @scope else @parentScope
-        if not scope? then throw 'No scope or parentScope set!'
-        GmapUtil.equalCoords @evalModelHandle(model1, scope.coords),
-          @evalModelHandle(model2, scope.coords)
+        hasCoords = _.contains(@interface.scopeKeys, 'coords')
+        scope = @scope if hasCoords and  @scope.coords? or not hasCoords
+        if not scope? then throw 'No scope set!'
+        if hasCoords
+          isEqual = GmapUtil.equalCoords @evalModelHandle(model1, scope.coords),
+            @evalModelHandle(model2, scope.coords)
+          #deep comparison of the rest of properties
+          return isEqual unless isEqual
+        #compare the rest of the properties that are being watched by scope
+        isEqual = _.every _.without(@interface.scopeKeys, 'coords'), (k) =>
+          @evalModelHandle(model1, scope[k]) == @evalModelHandle(model2, scope[k])
```

Comment:
This is the meat of the change. We no longer default to transcluded (window-child-model logic). Iterate through all scope properties to see if something is different.